import-scripts/verify-management-state.sh

```diff
@@ -10,42 +10,79 @@
 # - capture the current database color according to the management database table
 # - exit with error if they differ
 
-unset BLUE_SERVICE_LIST
-unset GREEN_SERVICE_LIST
-declare -a BLUE_SERVICE_LIST
-declare -a GREEN_SERVICE_LIST
-BLUE_SERVICE_LIST+=('cbioportal-backend-public-blue')
-BLUE_SERVICE_LIST+=('cbioportal-backend-public-beta-blue')
-BLUE_SERVICE_LIST+=('cbioportal-backend-master-blue')
-BLUE_SERVICE_LIST+=('cbioportal-backend-clickhouse-only-db-blue')
-GREEN_SERVICE_LIST+=('cbioportal-backend-public-green')
-GREEN_SERVICE_LIST+=('cbioportal-backend-public-beta-green')
-GREEN_SERVICE_LIST+=('cbioportal-backend-master-green')
-GREEN_SERVICE_LIST+=('cbioportal-backend-clickhouse-only-db-green')
-PUBLIC_CLUSTER_INGRESS_NAME='cbioportal-ingress'
-tmp="/data/portal-cron/tmp/import-cron-public"
+declare -a BLUE_SERVICE_LIST=()
+declare -a GREEN_SERVICE_LIST=()
+YQ_BINARY=${YQ_BINARY:-yq}
+
+function read_scalar() {
+    local key="$1"
+    local value
+    value=$("$YQ_BINARY" -r "$key" "$BLUEGREEN_CONFIG_FILEPATH")
+    if [ "$value" == "null" ] || [ -z "$value" ] ; then
+        echo "Error : missing required scalar '$key' in $BLUEGREEN_CONFIG_FILEPATH" >&2
+        exit 1
+    fi
+    printf '%s\n' "$value"
+}
+
+function read_array() {
+    local dest_var="$1"
+    local path="$2"
+    local array_path="${path}[]"
+    local file="$BLUEGREEN_CONFIG_FILEPATH"
+    local type=$("$YQ_BINARY" -r "$path | type" "$file")
+    if [ "$type" != "!!seq" ] ; then
+        echo "Error : expected array at '$path' in $file" >&2
+        exit 1
+    fi
+    unset "$dest_var"
+    declare -g -a "$dest_var"
+    readarray -t "$dest_var" < <("$YQ_BINARY" -r "$array_path" "$file")
+}
+
+function load_bluegreen_config() {
+    if ! [ -f "$BLUEGREEN_CONFIG_FILEPATH" ] || ! [ -r "$BLUEGREEN_CONFIG_FILEPATH" ] ; then
+        echo "Error : unable to read config file '$BLUEGREEN_CONFIG_FILEPATH'" >&2
+        exit 1
+    fi
+    if ! command -v "$YQ_BINARY" >/dev/null 2>&1 ; then
+        echo "Error : unable to locate yq binary '$YQ_BINARY' required to read config file" >&2
+        exit 1
+    fi
+    SERVICE_ACCOUNT=$(read_scalar '.service_account')
+    CLUSTER_KUBECONFIG=$(read_scalar '.cluster_cfg')
+    TEMP_DIR_PATH=$(read_scalar '.temp_dir_path')
+    INGRESS_NAME=$(read_scalar '.ingress_name')
+    read_array BLUE_SERVICE_LIST '.blue_deployment_list'
+    read_array GREEN_SERVICE_LIST '.green_deployment_list'
+    mkdir -p "$TEMP_DIR_PATH"
+}
 
 function usage() {
-    echo "usage: import-public-verify-management-state.sh cluster-management-file"
+    echo "usage: verify-management-state.sh bluegreen-config-file cluster-management-file"
```

Comment:
elsewhere I make the comment that I had a convention of putting the cluster-management-file as the first argument across many import scripts. I would prefer to keep to this convention. Also, maybe the two configuration files should be combined.